src/ray/core_worker/actor_manager.h

```diff
@@ -196,6 +212,15 @@ class ActorManager {
   /// getting them from GCS frequently.
   absl::flat_hash_map<std::string, ActorID> cached_actor_name_to_ids_
       GUARDED_BY(cache_mutex_);
+
+  /// Map from actor id to it's state(true: valid, false: invalid).
+  /// The state of actor is true When the actor is out of scope or is killed
+  absl::flat_hash_map<ActorID, bool> subscribed_actors_;
+  /// Protects access `subscribed_actors_`.
+  mutable absl::Mutex subscription_mutex_;
```

Comment:
Could we use a single mutex for both the cache and subscription? I think we will often be hitting these at the same time anyway, so it's simpler to use one.

Reply:
Okey, It's no problem.